### PR TITLE
Fix the log messages and log them at a higher debugging level

### PR DESCRIPTION
**Pull request by npadole20**
#### Describe Bug or Feature

As part of https://github.com/PBSPro/pbspro/pull/1737, log messages were added as log error to mom_mach.c, win_remote_shell.c. Some of them are flooding the mom logs even when there is no fatal error. 

#### Describe Your Change

1. mom_mach.c: removed the offending message, moved some of the log messages to DEBUG3
2. win_remote_shell.c: removed the offending message, moved some of the log messages to DEBUG3
3. vnode_storage.c: As part of PR #1737, I had reverted the error code for add_vmapent_byID() based on the function header. Based on the call path new_vnid() ->  add_mominfo() ->  add_vmapent_byID(), the return values are correct. So reverted those changes (kept the refactored code for add_vmapent_byID() as is) and updated the function header to reflect correct success and failure values.

#### Attach Test and Valgrind Logs/Output

Before fix:
[before_logs.txt](https://github.com/openpbs/openpbs/files/4716709/before_logs.txt)

After fix:
1. 
```
06/01/2020 09:27:04;0006;pbs_mom;Fil;pbs_mom;Version 2021.0.0.20200531154923, started, initialization type = 0
06/01/2020 09:27:04;0002;pbs_mom;Svr;pbs_mom;Mom pid = 3048 ready, using ports Server:15001 MOM:15002 RM:15003
06/01/2020 09:27:05;0002;pbs_mom;n/a;initialize;pcpus=2, OS reports 2 cpu(s)
06/01/2020 09:27:06;0d80;pbs_mom;TPP;pbs_mom(Main Thread);net restore handler called
06/01/2020 09:27:06;0002;pbs_mom;Svr;pbs_mom;Restart sent to server at dobby:15001
06/01/2020 09:27:06;0002;pbs_mom;Svr;pbs_mom;Hello from server at 192.168.153.144:15001
06/01/2020 09:27:08;0100;pbs_mom;Req;;Type 85 request received from root@192.168.153.144:15001, sock=1
06/01/2020 09:27:08;0080;pbs_mom;Hook;PBS_power.HK;copy hook-related file request received
```

2. 
```
06/02/2020 09:48:18;0008;pbs_mom;Job;2.dobby;Started, pid = 12132
06/02/2020 09:48:18;0008;pbs_mom;Job;2.dobby;Started, pid = 12132
06/02/2020 09:48:18;0001;pbs_mom;Svr;secure_file;Path C:/PROGRA~2/PBS/home/mom_priv/jobs/2.dobby.JC don't exists
06/02/2020 09:48:18;0001;pbs_mom;Svr;secure_file;Path C:/PROGRA~2/PBS/home/mom_priv/jobs/2.dobby.JB don't exists
06/02/2020 09:48:32;0080;pbs_mom;Job;2.dobby;task 1 terminated
06/02/2020 09:48:32;0008;pbs_mom;Job;2.dobby;Terminated
06/02/2020 09:48:32;0100;pbs_mom;Job;2.dobby;task 00000001 cput=00:00:00
06/02/2020 09:48:32;0008;pbs_mom;Job;2.dobby;kill_job
06/02/2020 09:48:32;0100;pbs_mom;Job;2.dobby;winclient cput=00:00:00 mem=858993456kb
06/02/2020 09:48:32;0100;pbs_mom;Job;2.dobby;Obit sent
06/02/2020 09:48:32;0100;pbs_mom;Req;;Type 6 request received from root@192.168.153.144:15001, sock=1
06/02/2020 09:48:32;0080;pbs_mom;Job;2.dobby;delete job request received
06/02/2020 09:48:32;0008;pbs_mom;Job;2.dobby;kill_job
```
[qsub.txt](https://github.com/openpbs/openpbs/files/4847445/qsub.txt)
[qsub_I.txt](https://github.com/openpbs/openpbs/files/4847447/qsub_I.txt)
[vnode.txt](https://github.com/openpbs/openpbs/files/4847449/vnode.txt)

### CODE REVIEW

**Comment by SudeshnaMoulik** [comment]
Can you please add the mom logs before fix and after fix ?

File: src/resmom/vnode_storage.c [hunk 1]
```diff
@@ -152,25 +152,25 @@ find_vmapent_byID(void *ctx, const char *vnid)
  * @param[in] data - information about vnode
  *
  * @return 	int
- * @retval 	0	Success
- * @retval 	1	Failure
+ * @retval 	1	Sucess
+ * @retval 	0	Failure
```
[the remark this answers is not in the document]
Comment (npadole20) [reply src/resmom/vnode_storage.c:156]: As part of PR #1737, I had reverted the error code for add_vmapent_byID() based on the function header. Based on the call path new_vnid() -> add_mominfo() -> add_vmapent_byID(), the return values are correct. So reverted those changes (kept the refactored code for add_vmapent_byID() as is) and updated the function header to reflect correct success and failure values. 
In case of failure we are returning zero in add_vmapent_byID()

File: src/resmom_win/win/mom_mach.c [hunk 2]
```diff
@@ -540,13 +540,10 @@ mem_sum(job *pjob)
 	GetSystemInfo(&si);
 
 	/* Get the number of processes embedded in the job */
-	if (!(ret = QueryInformationJobObject(pjob->ji_hJob,
-			JobObjectBasicAccountingInformation,
-			&ji, sizeof(ji), NULL))) {
-		log_errf(-1, __func__, "QueryInformationJobObject for %d failed to get number of processes", pjob->ji_hJob);
```
[the remark this answers is not in the document]
Comment (npadole20) [reply src/resmom_win/win/mom_mach.c:546]: QueryInformationJobObject this code is called from mom main thread among otehr places. It will be called even when there are no jobs as per my understanding. So we will be logging errors wrongly. Same for below.

[the remark this answers is not in the document]
Comment (SudeshnaMoulik) [reply src/resmom_win/win/mom_mach.c:546]: I think we call QueryInformationJobObject with a job handle. In-case if you noticed there is a scenario where we called this without a proper job, then we can get that validated from job id of that pjob.
Apart from this, if we look cput_job in mom_mach.c, there we do log the failure if the call fails.

File: src/resmom/vnode_storage.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (SudeshnaMoulik) [reply src/resmom/vnode_storage.c:156]: Do you think we should update add_mominfo header too ? Because it says it return 0 for success, which may contradict now ?

**Comment by npadole20** [comment]
[logs_after_review_comments.txt](https://github.com/openpbs/openpbs/files/4727536/logs_after_review_comments.txt)

**Comment by SudeshnaMoulik** [comment]
Can you please sync with the master branch and attach the high debug log ? In that way will not get those error logged which are already fixed and can also verify what actually getting logged in DEBUG3

**Comment by npadole20** [comment]
> 
> 
> Can you please sync with the master branch and attach the high debug log ? In that way will not get those error logged which are already fixed and can also verify what actually getting logged in DEBUG3

The mom logs attached has debug logs for 3 debug levels - normal, 1024 and 4095.

File: src/resmom_win/win/mom_mach.c [hunk 3]
```diff
@@ -536,18 +536,19 @@ mem_sum(job *pjob)
...
-	}
-
-	if ((pjob->ji_hJob != NULL) && ret) {
-		nps = ji.TotalProcesses;
+			&ji, sizeof(ji), NULL))
+			nps = ji.TotalProcesses;
+		else
+			log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
+				"QueryInformationJobObject for %d failed to get number of processes with error %lu", pjob->ji_hJob, GetLastError());
```
Comment (SudeshnaMoulik) [inline src/resmom_win/win/mom_mach.c:551]: Instead of logging job handle, do you think it will be better to print job id?

File: src/resmom_win/win/mom_mach.c [hunk 4]
```diff
@@ -571,11 +572,11 @@ mem_sum(job *pjob)
...
 	if (pjob->ji_hJob != NULL) {
-		ret = QueryInformationJobObject(pjob->ji_hJob,
+		if (!QueryInformationJobObject(pjob->ji_hJob,
 			JobObjectBasicProcessIdList,
-			pProcessList, pidlistsize, NULL);
-		if (!ret)
-			log_errf(-1, __func__, "QueryInformationJobObject for %d failed to get number of processes", pjob->ji_hJob);
+			pProcessList, pidlistsize, NULL))
+			log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
+				"QueryInformationJobObject for %d failed to get processe ID list with error %lu", pjob->ji_hJob, GetLastError());
```
Comment (SudeshnaMoulik) [inline src/resmom_win/win/mom_mach.c:579]: Instead of logging job handle, do you think it will be better to log job id ?

File: src/resmom_win/win/mom_mach.c [hunk 5]
```diff
@@ -644,24 +649,26 @@ cput_sum(job *pjob)
...
+	if (pjob->ji_hJob != NULL) {
+		if (QueryInformationJobObject(pjob->ji_hJob,
+			JobObjectBasicAccountingInformation,
+			&ji, sizeof(ji), NULL)) {
+			cputime = (double)(ji.TotalUserTime.QuadPart +
+				ji.TotalKernelTime.QuadPart);
+			nps = ji.TotalProcesses;
+		} else
+			log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
+				"QueryInformationJobObject for %d failed to get number of processes with error %lu", pjob->ji_hJob, GetLastError());
```
Comment (SudeshnaMoulik) [inline src/resmom_win/win/mom_mach.c:662]: Please log job id instead of job handle

File: src/resmom_win/win/mom_mach.c [hunk 6]
```diff
@@ -548,7 +548,7 @@ mem_sum(job *pjob)
 			nps = ji.TotalProcesses;
 		else
 			log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
-				"QueryInformationJobObject for %d failed to get number of processes with error %lu", pjob->ji_hJob, GetLastError());
+				"QueryInformationJobObject for job %d, handle %d failed to get number of processes with error %lu", pjob->ji_qs.ji_jobid, pjob->ji_hJob, GetLastError());
```
Comment (Bhagat-Rajput) [inline src/resmom_win/win/mom_mach.c:551]: pjob->ji_qs.ji_jobid, it's a char buffer so please change your format specifier for it.

File: src/resmom_win/win/mom_mach.c [hunk 7]
```diff
@@ -576,7 +576,7 @@ mem_sum(job *pjob)
 			JobObjectBasicProcessIdList,
 			pProcessList, pidlistsize, NULL))
 			log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
-				"QueryInformationJobObject for %d failed to get processe ID list with error %lu", pjob->ji_hJob, GetLastError());
+				"QueryInformationJobObject for job %d, handle %d failed to get processe ID list with error %lu", pjob->ji_qs.ji_jobid, pjob->ji_hJob, GetLastError());
```
Comment (Bhagat-Rajput) [inline src/resmom_win/win/mom_mach.c:579]: change your format specifier here and some other places as well wherever you are trying to log jobid

File: src/resmom_win/win/mom_mach.c [hunk 8]
```diff
@@ -610,7 +610,7 @@ mem_sum(job *pjob)
 			ret = IsProcessInJob(ptask->ti_hProc, pjob->ji_hJob, &is_process_in_job);
 			if (!ret)
 				log_eventf(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SERVER, LOG_DEBUG, __func__,
-					"IsProcessInJob failed for %d job %d process with errno %lu", pjob->ji_hJob, ptask->ti_hProc, GetLastError());
+					"IsProcessInJob failed for %s job %d process with errno %lu", pjob->ji_hJob, ptask->ti_hProc, GetLastError());
```
Comment (Bhagat-Rajput) [inline src/resmom_win/win/mom_mach.c:613]: why did you change here? Please check your format message because as per message i guess you are trying to print the jobid but passing the handle value to it.

File: src/resmom_win/win/mom_mach.c [hunk 9]
```diff
@@ -668,7 +668,7 @@ cput_sum(job *pjob)
...
```
Comment (Bhagat-Rajput) [inline src/resmom_win/win/mom_mach.c:671]: check your log message and pass the corresponding values to it with appropriate format specifier.

File: src/resmom_win/win/mom_mach.c [hunk 10]
```diff
@@ -1183,7 +1183,7 @@ cput_job(char *jobid)
...
```
Comment (Bhagat-Rajput) [inline src/resmom_win/win/mom_mach.c:1186]: same comment for here as well

File: src/lib/Libwin/win_remote_shell.c [hunk 11]
```diff
@@ -641,8 +641,6 @@ handle_stdoe_pipe(HANDLE hPipe_remote_std, void (*oe_handler)(char*))
 		if (!ReadFile(hPipe_remote_std, readbuf, READBUF_SIZE, &dwRead, NULL) || dwRead == 0) {
 			dwErr = GetLastError();
 			if (dwErr == ERROR_NO_DATA) {
-				log_eventf(PBSEVENT_ERROR, PBS_EVENTCLASS_SERVER, LOG_ERR,
```
Comment (subhasisb) [inline src/lib/Libwin/win_remote_shell.c:644]: Is this not a true error? Why should we NOT log this?

File: src/resmom_win/win/mom_mach.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (subhasisb) [reply src/resmom_win/win/mom_mach.c:546]: I guess the case would be when the job has no task running inside it. I think it might safest to leave this in DEBUG3

File: src/resmom_win/win/mom_mach.c [hunk 12]
```diff
@@ -570,13 +567,11 @@ mem_sum(job *pjob)
...
 
 	/* Get the pid list */
-	if (pjob->ji_hJob != NULL) {
-		ret = QueryInformationJobObject(pjob->ji_hJob,
+	if (pjob->ji_hJob != NULL)
+		QueryInformationJobObject(pjob->ji_hJob,
 			JobObjectBasicProcessIdList,
 			pProcessList, pidlistsize, NULL);
-		if (!ret)
-			log_errf(-1, __func__, "QueryInformationJobObject for %d failed to get number of processes", pjob->ji_hJob);
```
[the remark this answers is not in the document]
Comment (subhasisb) [reply src/resmom_win/win/mom_mach.c:578]: it is there as DEBUG3

**Comment by subhasisb** [comment]
Windows is not supported by OpenPBS, hence closing this PR as not relevant

**Comment by subhasisb** [comment]
Windows is not supported by OpenPBS, hence closing this PR as not relevant
